pulp_docker/app/viewsets.py

```diff
@@ -0,0 +1,107 @@
+"""
+Check `Plugin Writer's Guide`_ and `pulp_example`_ plugin
+implementation for more details.
+
+.. _Plugin Writer's Guide:
+    http://docs.pulpproject.org/en/3.0/nightly/plugins/plugin-writer/index.html
+
+.. _pulp_example:
+    https://github.com/pulp/pulp_example/
+"""
+
+from pulpcore.plugin import viewsets as core
+from pulpcore.plugin.serializers import (
+    RepositoryPublishURLSerializer,
+    RepositorySyncURLSerializer,
+)
+from pulpcore.plugin.tasking import enqueue_with_reservation
+from rest_framework.decorators import detail_route
+
+from . import models, serializers, tasks
+
+
+class DockerContentViewSet(core.ContentViewSet):
+    """
+    A ViewSet for DockerContent.
+
+    Define endpoint name which will appear in the API endpoint for this content type.
+    For example::
+        http://pulp.example.com/api/v3/content/docker/
+
+    Also specify queryset and serializer for DockerContent.
+    """
+
+    endpoint_name = 'docker'
+    queryset = models.DockerContent.objects.all()
+    serializer_class = serializers.DockerContentSerializer
+
+
+class DockerRemoteViewSet(core.RemoteViewSet):
+    """
+    A ViewSet for DockerRemote.
+
+    Similar to the DockerContentViewSet above, define endpoint_name,
+    queryset and serializer, at a minimum.
+    """
+
+    endpoint_name = 'docker'
+    queryset = models.DockerRemote.objects.all()
+    serializer_class = serializers.DockerRemoteSerializer
+
+    @detail_route(methods=('post',), serializer_class=RepositorySyncURLSerializer)
+    def sync(self, request, pk):
+        """
+        Synchronizes a repository. The ``repository`` field has to be provided.
+        """
+        remote = self.get_object()
+        serializer = RepositorySyncURLSerializer(data=request.data, context={'request': request})
+
+        # Validate synchronously to return 400 errors.
+        serializer.is_valid(raise_exception=True)
+        repository = serializer.validated_data.get('repository')
+        result = enqueue_with_reservation(
+            tasks.synchronize,
+            [repository, remote],
+            kwargs={
+                'remote_pk': remote.pk,
+                'repository_pk': repository.pk
+            }
+        )
+        return core.OperationPostponedResponse(result, request)
+
+
+class DockerPublisherViewSet(core.PublisherViewSet):
+    """
+    A ViewSet for DockerPublisher.
+
+    Similar to the DockerContentViewSet above, define endpoint_name,
+    queryset and serializer, at a minimum.
+    """
+
+    endpoint_name = 'docker'
+    queryset = models.DockerPublisher.objects.all()
+    serializer_class = serializers.DockerPublisherSerializer
+
+    @detail_route(methods=('post',), serializer_class=RepositoryPublishURLSerializer)
```

Comment:
Please add decorator for AsyncOperationReponse: 
https://github.com/pulp/plugin_template/blob/master/pulp_plugin_template/app/viewsets.py#L94